client/src/components/AddNewProfile.js

```diff
@@ -16,46 +17,96 @@ const AddNewProfile = () => {
     support_type: "",
     profile_type: "",
     status: "new",
-    join_date: new Date()
+    nationality: "",
+    join_date: new Date(),
   });
-
+  const [errors, setErrors] = useState([]);
   const [profileCreated, setProfileCreated] = useState(null);
+  const [nationality, setNationality] = useState([]);
 
-  const handleChange = event => {
+  const handleChange = (event) => {
     updateField(event.target.name, event.target.value);
   };
 
   const createProfile = () => {
-    postProfile(profileData).then(isSuccessful => {
+    let errs = [];
+    if (profileData.firstname.length === 0) {
+      errs.push(
+        <li>
+          <p>First name cannot be empty</p>
+        </li>
+      );
+    }
+    if (profileData.lastname.length === 0) {
+      errs.push(
+        <li>
+          <p>Last name cannot be empty</p>
+        </li>
+      );
+    }
+    if (profileData.phone.length < 10) {
+      errs.push(
+        <li>
+          <p>Phone number must have at least 10 digits</p>
+        </li>
+      );
+    }
+    if (errs.length > 0) {
+      setErrors(errs);
+      return;
+    }
+    postProfile(profileData).then((isSuccessful) => {
       setProfileCreated(isSuccessful);
     });
+    setErrors([]);
   };
-
+  
   const handleDropdownAndDateChange = (event, data) => {
     updateField(data.name, data.value);
   };
 
   const updateField = (name, value) => {
     const updatedProfileData = {
       ...profileData,
-      [name]: value
+      [name]: value,
     };
     setProfileData(updatedProfileData);
   };
 
   const profileTypeOptions = [
     { key: "volunteer", value: "volunteer", text: "Volunteer" },
-    { key: "service user", value: "service_user", text: "Service user" }
+    { key: "service user", value: "service_user", text: "Service user" },
   ];
 
   const statusOptions = [
     { key: "new", value: "new", text: "New" },
     { key: "active", value: "active", text: "Active" },
-    { key: "inactive", value: "inactive", text: "Inactive" }
+    { key: "inactive", value: "inactive", text: "Inactive" },
   ];
 
+  const genderOptions = [
+    { key: "male", value: "male", text: "Male" },
+    { key: "female", value: "female", text: "Female" },
+    { key: "other", value: "other", text: "Other" },
+    { key: "not_provided", value: "not_provided", text: "Not provided" },
+  ];
+
+  useEffect(() => {
+    getNationalities().then((response) => {
+      setNationality(response);
+    });
+  }, []);
+
+  const nationalityOptions = nationality.map((nationality) => ({
+    key: nationality.id,
+    text: nationality.nationality,
+    value: nationality.nationality,
+  }));
+
   return (
     <Form onSubmit={createProfile}>
+      <ul class="errors">{errors}
+      </ul>
```

Comment:
```suggestion
      <ul class="errors">
        {errors.map(error => <li><p>{error}</p></li>)}
      </ul>
```
We can do something like this to make a list of errors if we update the other code to put strings in `errors`, i.e. `errs.push('First name cannot be empty');`